tasks/steps/update_instance.yml

```diff
@@ -1,11 +1,20 @@
 ---
 
-- name: 'Update instance'
-  file:
-    src: '{{ instance_info.dist_dir }}'
-    dest: '{{ instance_info.instance_dist_dir }}'
-    owner: '{{ cartridge_app_user }}'
-    group: '{{ cartridge_app_group }}'
-    state: link
-  when: cartridge_multiversion
+- when: cartridge_multiversion
   tags: cartridge-instances
+  block:
+    - name: 'Update instance application'
+      file:
+        src: '{{ instance_info.dist_dir }}'
+        dest: '{{ instance_info.instance_dist_dir }}'
+        owner: '{{ cartridge_app_user }}'
+        group: '{{ cartridge_app_group }}'
+        state: link
+
+    - name: 'Check if instance restart is required to update application'
+      cartridge_set_needs_restart:
+        console_sock: '{{ instance_info.console_sock }}'
+        instance_dist_dir: '{{ instance_info.instance_dist_dir }}'
```

Comment:
It can be useful to pass `restarted` to this function: if restarted is not none we can simply return `needs_restart: null`

Reply:
Then the user will not be able to understand `needs_restart == true`, because there is a restarted flag or because the code/config has been updated